code/modules/projectiles/guns/projectile/launchers.dm

```diff
@@ -37,4 +37,108 @@
 /obj/item/weapon/gun/projectile/automatic/gyropistol/update_icon()
 	..()
 	icon_state = "[initial(icon_state)][magazine ? "loaded" : ""]"
-	return
\ No newline at end of file
+	return
+/obj/item/weapon/gun/buttlauncher
+	name = "Butt Launcher"
+	desc = "For HONKing on an extreme level!"
+	icon = 'icons/obj/buttlauncher.dmi'
+	icon_state = "buttempty"
+	item_state = "riotgun"
+	w_class = 2.0
```

Comment:
This is not a proper weight class for this item. It should be at the very least 3 or 4. Whichever one doesn't allow it in the backpack due to bulky item state is preferable.